logger.h

```diff
@@ -1,62 +1,129 @@
-  
+
 Stay tuned using
-twitter @navitia 
+twitter @navitia
 IRC #navitia on freenode
 https://groups.google.com/d/forum/navitia
 www.navitia.io
 */
-
 #pragma once
+
 #include "conf.h"
+#include <boost/format.hpp>
+
 #include <log4cplus/logger.h>
 #include <log4cplus/configurator.h>
-#include <boost/format.hpp>
 #ifdef HAVE_LOGGINGMACROS_H
 #include <log4cplus/loggingmacros.h>
 #endif
 
-/** Crée une configuration par défaut pour le logger */
-inline void init_logger(){
+namespace navitia {
+
+// logger pattern example : [18-09-10 17:02:17,186] [comment] [INFO ] [NDC] - message file.cpp:248
+static const std::string console_pattern_ref1 = "[%D{%y-%m-%d %H:%M:%S,%q}] ";
+static const std::string console_pattern_ref2 = "[%-5p] [%x] - %m %b:%L  %n";
+
+// syslog pattern example : Sep 11 14:16:11 hostname binary_name: [comment] [PID] [INFO] [NDC] - message file.cpp:261
+static const std::string syslog_pattern_ref  = "[%i] [%p] [%x] - %m %b:%L  %n";
+
+/**
+ * @brief init simple navitia logger
+ */
+inline void init_logger()
+{
     log4cplus::BasicConfigurator config;
     config.configure();
+
+    auto properties = log4cplus::helpers::Properties();
+    properties.setProperty("log4cplus.rootLogger", "DEBUG, console");
+    properties.setProperty("log4cplus.appender.console", "log4cplus::ConsoleAppender");
+    properties.setProperty("log4cplus.appender.console.layout", "log4cplus::PatternLayout");
+    properties.setProperty("log4cplus.appender.console.layout.ConversionPattern", console_pattern_ref1 + console_pattern_ref2);
+
+    log4cplus::PropertyConfigurator configurator(properties);
+    configurator.configure();
 }
 
-/** configure the logger with a level and a pattern, the output will be on stdout */
-inline void init_logger(const std::string& level, const std::string& pattern){
+/**
+ * @brief init navitia logger
+ *
+ * @param name The application name
+ * @param level The level of the logger (INFO, DEBUG, etc..)
+ * @param active_local_syslog Active syslog redirection
+ * @param pattern Overload pattern logger
+ */
+inline void init_logger(const std::string& name,
+                        const std::string& level,
+                        const bool active_local_syslog = false,
+                        const std::string& comment = "",
+                        const std::string& pattern = "")
+{
+    std::string prefix = "";
+    if (!comment.empty()) {
+        prefix = prefix + "[" + comment + "] ";
+    }
+
     log4cplus::BasicConfigurator config;
     config.configure();
+
     auto properties = log4cplus::helpers::Properties();
-    properties.setProperty("log4cplus.rootLogger", (boost::format("%s, default") % level).str());
-    properties.setProperty("log4cplus.appender.default", "log4cplus::ConsoleAppender");
-    properties.setProperty("log4cplus.appender.default.layout", "log4cplus::PatternLayout");
-    properties.setProperty("log4cplus.appender.default.layout.ConversionPattern", pattern);
+
```

Comment:
Minor: To make thinkgs more readable, you could define your patterns early with: 
```cpp
auto syslogPattern = prefix + syslog_pattern_ref;
auto consolePattern = console_pattern_ref1 + prefix + console_pattern_ref2;

if (pattern.empty()) {
    syslogPattern = prefix + pattern;
    consolePattern = prefix + pattern;
}
```

And use them later on without the `if(pattern.empty())` condition.

Reply:
You are slightly punctilious @nonifier haha.

Reply:
I know, I'm nitpicking :P